src/components/app/functions/FunctionsCard.tsx

```diff
@@ -0,0 +1,117 @@
+'use client';
+import {useEffect, useState} from 'react';
+
+import {RealtimeResponseEvent} from 'appwrite';
+
+import {AppwriteClient} from '@/api/config/client.config';
+import {EnvConfig} from '@/api/config/env.config';
+import {
+	createDestination,
+	deleteDestination,
+	getDestinationList,
+} from '@/api/modules/destinations';
+import {useScopedI18n} from '@/locales/client';
+import {Destination} from '@/models/destination';
+import {EventType, getEventType} from '@/utils/realtime.utils';
+
+export default function FunctionsCard() {
+	const t = useScopedI18n('functions');
+
+	const [destinations, setDestinations] = useState<Destination[]>([]);
+	const [isLoading, setIsLoading] = useState(false);
+
+	useEffect(() => {
+		getDestinationList().then((data) => {
+			setDestinations(data);
+		});
+
+		const unsubscribe = AppwriteClient.subscribe(
+			`databases.${EnvConfig.databaseId}.collections.${EnvConfig.destinationCollectionId}.documents`,
+			async (response: RealtimeResponseEvent<Destination>) => {
+				if (getEventType({events: response.events}) == EventType.DELETE) {
+					const distantDestinations = await getDestinationList();
+
+					setDestinations((oldDestinations) => {
+						const newDestinations = [...oldDestinations];
+						const oldDestinationsId = oldDestinations.map(
+							(oldDestination) => oldDestination.$id
+						);
+
+						const newDestination = distantDestinations.filter(
+							(distantDestination) =>
+								!oldDestinationsId.includes(distantDestination.$id)
+						)[0];
+
+						newDestinations.pop();
+						newDestinations.unshift(newDestination);
+
+						return newDestinations;
+					});
```

Comment:
Comme ça je dirais que ça fait pareil mais faut tester 😅

```suggestion
					setDestinations((oldDestinations) => {
					  const oldDestinationsId = oldDestinations.map((oldDestination) => oldDestination.$id);
					
					  const newDestination = distantDestinations.find(
					    (distantDestination) => !oldDestinationsId.includes(distantDestination.$id)
					  );
					
					  return [newDestination, ...oldDestinations.slice(0, -1)];
					});
```